libs/designsystem/menu/src/menu.component.ts

```diff
@@ -75,33 +82,193 @@ export class MenuComponent implements AfterViewInit, OnDestroy {
   public buttonContainerElement: ElementRef<HTMLElement> | undefined;
 
   @ViewChild('defaultButton', { read: ElementRef })
-  public defaultButtonElement: ElementRef<HTMLElement> | undefined;
+  public defaultButtonElement: ElementRef<HTMLButtonElement> | undefined;
 
   @ContentChild(ButtonComponent, { read: ElementRef }) public userProvidedButton:
-    | ElementRef<HTMLElement>
+    | ElementRef<HTMLButtonElement>
     | undefined;
 
   @ViewChild(FloatingDirective)
-  private floatingDirective: FloatingDirective;
+  _floatingMenu: FloatingDirective;
```

Comment:
We ususally only use this notation for properties that we need to make public to access in the template, but shouldn't be used outside the component. Consider reverting to private and use regular naming:
```suggestion
  private floatingMenu: FloatingDirective;
```